enhancements/kata-containers/kata-containers-dev-preview.md

```diff
@@ -0,0 +1,276 @@
+---
+title: kata-containers-dev-preview
+authors:
+  - "@aadam"
+reviewers:
+  - "@mrunalp"
+approvers:
+  - "@mrunalp"
+creation-date: 2020-06-07
+last-updated: 2020-06-07
+status: implementable
+see-also:
+  - TBD
+replaces:
+  - TBD
+superseded-by:
+  - TBD
+---
+
+# Kata Containers
+
+## Summary
+
+[Kata Containers](https://katacontainers.io/) is an open source project developing a container runtime using virtual machines and providing the same look and feel as vanilla containers.
+By leveraging hardware virtualization technologies, Kata Containers provides powerful workload isolation compared to existing container solutions.
+
+We will be integrating Kata Containers into OpenShift to provide the ability to run kernel isolated containers for any workload which requires:
+- custom kernel tuning (sysctl, scheduler changes, cache tuning, etc);
+- custom kernel modules (out of tree, special arguments, etc);
+- exclusive access to hardware;
+- root privileges;
+- any other administrative privileges above and beyond what is secure in a shared kernel environment (regular runc).
+
+It should be noted that Kata Containers differ from Kubevirt:
+
+- Kubevirt aims to run **VM images** on OpenShift providing the VM with the look and feel of a virtual machine running as a legacy VM (CRD defining a Virtual Machine in Kubernetes)
+- Kata Containers aim to run **container images** on OpenShift in an isolated manner using virtualization tools (uses Runtime Class resource to choose Kata only for specific workloads which require kernel level isolation)
```

Comment:
Semi random but there's actually a 3rd class of thing in here, which is "containers that use virt" - what we do with https://github.com/coreos/coreos-assembler/ overlaps a lot with kata; we want to control qemu, not be inside qemu.

Reply:
That's interesting, and I wonder whether the minimalist QEMU we plan to use for the extensions could also benefit CoreOS Assembler.

Do you mind if we switch this conversation to the Slack, at some point soon, to try to sync on the usages and then, maybe, come up with a full document on how we plan to use QEMU as part of RHCOS, and hopefully agree in the minimal QEMU that could attend yours (CoreOS Assembler) and ours (Kata Containers) needs?

Reply:
Note that the qemu we use to build (and test) RHCOS (and FCOS) is included in the coreos-assembler container; it has nothing to do with the host, which is the opposite of kata.  It's closer to KubeVirt in a way except this usage is special in that we're not trying to run production VMs that e.g. expose networking.  We don't require any privileges, just KVM available.

I have a blurb on this here https://hackmd.io/8YCMjP_JTzu-vp7PrOjT0g that I need to stick somewhere.

Reply:
We don't need a minimal qemu really - our container image is already enormous anyways (2.5G+).

What may be good to share though is e.g. Go bindings for interacting with qemu - I touched on that in https://github.com/coreos/coreos-assembler/pull/1336#issuecomment-611759775